src/openrct2/world/map_generator/PngTerrainGenerator.cpp

```diff
@@ -199,6 +205,7 @@ namespace OpenRCT2::World::MapGenerator
         const uint8_t rangeIn = maxValue - minValue;
         const uint8_t rangeOut = (settings->heightmapHigh - settings->heightmapLow) * 2;
 
+        MapInit({ mapHeight, mapWidth });
```

Comment:
Why has this line been moved?

Reply:
For the progress bar. This way the refreshed screen area around the progress bar window will still show the previous terrain instead of the newly initialized map